function.rb

```diff
@@ -0,0 +1,9 @@
+
+def function(n)
+  if n == 0
+    1
```

Comment:
I think you need to explicitly return 1 here and, on line 6, return `n * function(n-1)`